stakingModel/test/SimpleDAIStaking.test.js

```diff
@@ -772,6 +766,128 @@ contract("SimpleDAIStaking - staking with DAI mocks", ([founder, staker]) => {
+    await cDAI.mint(web3.utils.toWei("100", "ether"));
+    const cdaiBalanceFounder = await cDAI.balanceOf(founder);
+    await cDAI.transfer(simpleStaking.address, cdaiBalanceFounder);
+    let avatarBalanceBefore = await cDAI.balanceOf(avatar.address);
+    let encodedCall = web3.eth.abi.encodeFunctionCall(
+      {
+        name: "recover",
+        type: "function",
+        inputs: [
+          {
+            type: "address",
+            name: "_token"
+          }
+        ]
+      },
+      [cDAI.address]
+    );
+    await controller.genericCall(simpleStaking.address, encodedCall, avatar.address, 0);
+    let avatarBalanceAfter = await cDAI.balanceOf(avatar.address);
+    expect(avatarBalanceAfter.sub(avatarBalanceBefore).toString()).to.be.equal(
+      web3.utils.toWei("0", "ether")
+    );
+  });
+
+  it("should not transfer excessive cdai funds when total staked is 0 and not paused and execute recover", async () => {
+    let simpleStaking1 = await SimpleDAIStaking.new(
+      dai.address,
+      cDAI.address,
+      founder,
+      BLOCK_INTERVAL,
+      avatar.address,
+      identity.address
+    );
+    let avatarBalanceBefore = await cDAI.balanceOf(avatar.address);
+    let encodedCall = web3.eth.abi.encodeFunctionCall(
+      {
+        name: "recover",
+        type: "function",
+        inputs: [
+          {
+            type: "address",
+            name: "_token"
+          }
+        ]
+      },
+      [cDAI.address]
+    );
+    await controller.genericCall(simpleStaking1.address, encodedCall, avatar.address, 0);
+    let avatarBalanceAfter = await cDAI.balanceOf(avatar.address);
+    expect(avatarBalanceAfter.sub(avatarBalanceBefore).toString()).to.be.equal(
+      web3.utils.toWei("0", "ether")
+    );
+  });
+
+  it("should transfer excessive cdai funds when execute recover", async () => {
+    await dai.mint(founder, web3.utils.toWei("100", "ether"));
+    await dai.approve(cDAI.address, web3.utils.toWei("100", "ether"));
+    const cdaiBalanceFounder1 = await cDAI.balanceOf(founder);
+    await cDAI.mint(web3.utils.toWei("100", "ether"));
+    const cdaiBalanceFounder2 = await cDAI.balanceOf(founder);
+    await cDAI.transfer(simpleStaking.address, cdaiBalanceFounder2.sub(cdaiBalanceFounder1).toString());
+    let avatarBalanceBefore = await cDAI.balanceOf(avatar.address);
+    await simpleStaking.withdrawStake({
+      from: staker
+    });
+    let encodedCall = web3.eth.abi.encodeFunctionCall(
+      {
+        name: "recover",
+        type: "function",
+        inputs: [
+          {
+            type: "address",
+            name: "_token"
+          }
+        ]
+      },
+      [cDAI.address]
+    );
+    await controller.genericCall(simpleStaking.address, encodedCall, avatar.address, 0);
+    let avatarBalanceAfter = await cDAI.balanceOf(avatar.address);
+    expect(avatarBalanceAfter.sub(avatarBalanceBefore).toString()).to.be.equal(
+      "19796231467"
```

Comment:
**numbers with explanation.**

1. just check it is not 0 (ie something was recovered)
2. verify that staking balance is 0